src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishASingleFileApp.cs

```diff
@@ -651,5 +651,59 @@ public void It_errors_when_including_symbols_targeting_net5(bool selfContained)
                 .And
                 .HaveStdOutContaining(Strings.CannotIncludeSymbolsInSingleFile);
         }
+
+        [RequiresMSBuildVersionFact("16.8.0")]
+        public void It_errors_when_enabling_compression_targeting_net5()
+        {
+            var testProject = new TestProject()
+            {
+                Name = "SingleFileTest",
+                TargetFrameworks = "net5.0",
+                IsExe = true,
+            };
+
+            testProject.AdditionalProperties.Add("EnableCompressionInSingleFile", "true");
+
+            var testAsset = _testAssetsManager.CreateTestProject(testProject);
+            var publishCommand = new PublishCommand(testAsset);
+
+            publishCommand.Execute(PublishSingleFile, RuntimeIdentifier)
+                .Should()
+                .Fail()
+                .And
+                .HaveStdOutContaining(Strings.CannotEnableCompressionInSingleFile);
+        }
+
+        [RequiresMSBuildVersionFact("16.8.0")]
+        public void It_compresses_single_file_by_default()
+        {
+            var testProject = new TestProject()
+            {
+                Name = "SingleFileTest",
+                TargetFrameworks = "net6.0",
+                IsExe = true,
+            };
+            testProject.AdditionalProperties.Add("SelfContained", "true");
+
+            var testAsset = _testAssetsManager.CreateTestProject(testProject);
+            var publishCommand = new PublishCommand(testAsset);
+            var singleFilePath = Path.Combine(GetPublishDirectory(publishCommand, "net6.0").FullName, $"SingleFileTest{Constants.ExeSuffix}");
+
+            publishCommand
+                .Execute(PublishSingleFile, RuntimeIdentifier, IncludeAllContent, "/p:EnableCompressionInSingleFile=false")
```

Comment:
We need to use `IncludeAllContent` here to be sure there is something to compress. Since we do not compress managed assemblies yet, we need some native libs in the bundle.
Once https://github.com/dotnet/runtime/pull/50817 is merged and changes propagate to SDK, we can drop `IncludeAllContent` , although that is not highly important since the main goal is to check that compression is enabled by default.